nym-vpn-core/nym-vpn-lib/src/credentials/helpers.rs

```diff
@@ -40,6 +41,49 @@ pub enum CredentialStoreError {
         path: PathBuf,
         source: std::io::Error,
     },
+
+    #[error("failed to copy old db file: {source}")]
+    FailedToLookupDbVersion {
+        #[from]
+        source: sqlx::Error,
+    },
+
+    #[error("failed to copy old db file: {0}")]
+    FailedToCopyOldDbFile(std::io::Error),
+}
+
+fn forked_db_path(db_path: &Path) -> PathBuf {
+    db_path.with_file_name(format!(
+        "fork_{}",
+        db_path.file_name().unwrap().to_str().unwrap()
+    ))
+}
+
+async fn is_db_old(db_path: &str) -> Result<bool, CredentialStoreError> {
+    let pool = SqlitePoolOptions::new()
+        .max_connections(5)
+        .connect(db_path)
+        .await?;
+    let row = sqlx::query("SELECT MAX(version) as version FROM _sqlx_migrations")
+        .fetch_one(&pool)
+        .await?;
+    let migration_version: i64 = row.get("version");
+    Ok(migration_version == 20241104120000)
```

Comment:
that's the coconut/freepass version, right?

Reply:
Yip! Extracted out as named constant